test/cpp/api/autograd.cpp

```diff
@@ -154,6 +155,39 @@ TEST(AutogradAPITests, RetainGrad) {
   ASSERT_VARIABLE_EQ(input * 18, input.grad());
 }
 
+TEST(AutogradAPITests, AnomalyMode) {
+  // Needs to have backtrace as warning and then throw an error
+  torch::autograd::AnomalyMode::set_enabled(true);
```

Comment:
nit: this is not a context manager in CPP right? We need to reset it to the old value at the end of the test?